internal/uvm/scsi/backend.go

```diff
@@ -45,14 +45,14 @@ func NewHCSAttacher(system *hcs.System) Attacher {
 	return &hcsAttacher{system}
 }
 
-func (ha *hcsAttacher) attach(ctx context.Context, controller, lun uint, config *attachConfig) error {
+func (ha *hcsAttacher) attach(ctx context.Context, controller, lun uint, config *AttachConfig) error {
 	req := &hcsschema.ModifySettingRequest{
 		RequestType: guestrequest.RequestTypeAdd,
 		Settings: hcsschema.Attachment{
-			Path:                      config.path,
-			Type_:                     config.typ,
-			ReadOnly:                  config.readOnly,
-			ExtensibleVirtualDiskType: config.evdType,
+			Path:                      config.Path,
+			Type_:                     config.Type,
```

Comment:
I think there's a build break here 

```
# github.com/Microsoft/hcsshim/internal/uvm/scsi
internal\uvm\scsi\backend.go:53:31: cannot use config.Type (variable of type AttachmentType) as string value in struct literal
```

Reply:
Yup, this is due to the experimental commit I wrote up to demonstrate the other Manager interface we could expose.